pkgs/development/python-modules/digitalocean/default.nix

```
@@ -1,23 +1,51 @@
-{ stdenv, buildPythonPackage, fetchPypi, requests, jsonpickle }:
+{ buildPythonPackage
+, fetchFromGitHub
+, fetchPypi
+, isPy3k
+, jsonpickle
+, mock
+, pytest
+, requests
+, responses
+, stdenv
+}:
 
 buildPythonPackage rec {
   pname = "python-digitalocean";
-  version = "1.13.2";
+  version = "1.15.0";
 
-  src = fetchPypi {
-    inherit pname version;
-    sha256 = "0h4drpdsmk0b3rlvg6q6cz11k23w0swj1iddk7xdcw4m7r7c52kw";
+  src = fetchFromGitHub {
+    name = "python-digitalocean-${version}";
+    owner = "koalalorenzo";
+    repo = "python-digitalocean";
+    rev = "v${version}";
+    sha256 = "1pz15mh72i992p63grwzqn2bbp6sm37zcp4f0fy1z7rsargwsbcz";
   };
 
-  propagatedBuildInputs = [ requests jsonpickle ];
+  propagatedBuildInputs = [
+    jsonpickle
+    requests
+  ];
 
-  # Package doesn't distribute tests.
-  doCheck = false;
+  checkInputs = [
+    pytest
+    responses
+  ] ++ stdenv.lib.optionals (!isPy3k) [
+    mock
+  ];
```

```suggestion
  checkInputs = [
    pytest
    responses
    pytestCheckHook
  ] ++ stdenv.lib.optionals (!isPy3k) [
    mock
  ];
```

just runs pytest nicely.

---

i tried this and it didn't work

---

that's because the `setuptoolsCheckPhase` is still being ran, you would either need to do `dontUseSetuptoolsCheck = true;` or, instead of changing directory, you could almost move/delete the digitalocean directory.

---

oh ok thanks. i updated it now